library/core/src/ptr/mod.rs

```diff
@@ -893,8 +895,9 @@ pub unsafe fn write<T>(dst: *mut T, src: T) {
     // `dst` cannot overlap `src` because the caller has mutable access
     // to `dst` while `src` is owned by this function.
     unsafe {
-        copy_nonoverlapping(&src as *const T, dst, 1);
-        intrinsics::forget(src);
+        // FIXME: No need for mutable borrow once #80418 hits beta
+        copy_nonoverlapping(&mut src as *const T, dst, 1);
+        mem::forget(src);
```

Comment:
Until #79989 is merged, using `mem::forget` over `intrinsics::forget` will likely undo a lot/some of the performance improvements due to the inlining done in #80290.

Reply:
FWIW, this is still a potential blocker.

Reply:
can we just keep using `intrinsics::forget` here? This is an orthogonal change to this PR, right?

Reply:
The reason I changed to `mem::forget` is because `intrinsics::forget` is not `const`

Reply:
oh... right. So... until we resolve that issue, maybe we could make the `forget` intrinsic const? What do you think @RalfJung ? The forget intrinsic is trivial, so it won't be an impl burden while we wait for debug mode optimizations to get a handle on maybeuninit and friends.

Reply:
https://github.com/rust-lang/rust/pull/79989 will have to make the intrinsic const anyway, so -- sure, works for me.

Reply:
Note that the impl burden is 0, the intrinsic is entirely implemented by the intrinsic lowering MIR pass. Miri does not have an implementation of this intrinsic any more (https://github.com/rust-lang/miri/pull/1658).

Reply:
Is there anything more than adding
```
#[rustc_const_unstable(feature = "const_intrinsic_forget", issue = "none")]
```
?

Reply:
Nope, that is good. I think we can leave off giving this an issue just like `unlikely`. This is going to get stabilized with #79989 anyway